test/cluster/test_truncate_concurrent_writes.py

```diff
@@ -0,0 +1,167 @@
+#
+# Copyright (C) 2025-present ScyllaDB
+#
+# SPDX-License-Identifier: LicenseRef-ScyllaDB-Source-Available-1.0
+#
+
+from test.pylib.manager_client import ManagerClient
+from test.pylib.rest_client import inject_error_one_shot, read_barrier
+from test.cluster.conftest import skip_mode
+from test.cluster.util import create_new_test_keyspace
+from cassandra.query import SimpleStatement, ConsistencyLevel
+
+import pytest
+import asyncio
+import logging
+import time
+import random
+import glob
+import os
+
+logger = logging.getLogger(__name__)
+
+@pytest.mark.asyncio
+@skip_mode('release', 'error injections are not supported in release mode')
+async def test_validate_truncate_with_concurrent_writes(manager: ManagerClient):
+    cmdline = []
+    config = {}
+
+    servers = await manager.servers_add(3, config=config, cmdline=cmdline, property_file=[
+        {"dc": "dc1", "rack": "r1"},
+        {"dc": "dc1", "rack": "r2"},
+        {"dc": "dc1", "rack": "r3"},
+    ])
+
+    cql = manager.get_cql()
+    ks = await create_new_test_keyspace(cql, f"WITH replication = {{'class': 'NetworkTopologyStrategy', 'replication_factor': 3}}")
+    await cql.run_async(f"CREATE TABLE {ks}.test (pk int, ck int, val int, PRIMARY KEY(pk, ck));")
+
+    trunc_started_event = asyncio.Event()
+    trunc_completed_event = asyncio.Event()
+    writer_halfpoint = asyncio.Event()
+
+    number_of_ck_to_insert = 10000
+    writer_results = {}
+    async def writer(pk, writer_halfpoint, trunc_started_event, trunc_completed_event):
+        stmt = cql.prepare(f"INSERT INTO {ks}.test (pk, ck, val) VALUES (?, ?, ?);")
+        stmt.consistency_level = ConsistencyLevel.ONE
+        last_ck_before_truncate = None
+        first_ck_after_truncate = None
+        logger.info(f'write of pk: {pk} started')
+        for ck in range(number_of_ck_to_insert):
+            if not trunc_started_event.is_set():
+                last_ck_before_truncate = ck
+
+            await cql.run_async(stmt, [pk, ck, ck])
+
+            if first_ck_after_truncate is None and trunc_completed_event.is_set():
+                first_ck_after_truncate = ck
+
+            if ck % 50 == 0:
+                logger.info(f'written pk: {pk} ck: {ck}')
+
+            if ck > number_of_ck_to_insert / 2:
+                writer_halfpoint.set()
+
+        writer_results[pk] = (last_ck_before_truncate, first_ck_after_truncate)
+        logger.info(f'write of pk: {pk} ended')
```

Comment:
nit: maybe add the values of `last_ck_before_truncate` and `first_ck_after_truncate` to the log message